benchmarks/entropy_adiabat/plugins/entropy_model.cc

```diff
@@ -0,0 +1,371 @@
+          max_temp=-std::numeric_limits<double>::max();
+          n_temperature=0;
+          n_pressure=0;
+
+          std::string temp;
+          // Read data from disk and distribute among processes
+          std::istringstream in(Utilities::read_and_distribute_file_content(filename, comm));
+
+          std::getline(in, temp); // eat first line
+          std::getline(in, temp); // eat next line
+          std::getline(in, temp); // eat next line
+          std::getline(in, temp); // eat next line
+
+          in >> min_temp;
+          std::getline(in, temp);
+          in >> delta_temp;
+          std::getline(in, temp);
+          in >> n_temperature;
+          std::getline(in, temp);
+          std::getline(in, temp);
+          in >> min_press;
+          min_press *= 1e5;  // conversion from [bar] to [Pa]
+          std::getline(in, temp);
+          in >> delta_press;
+          delta_press *= 1e5; // conversion from [bar] to [Pa]
+          std::getline(in, temp);
+          in >> n_pressure;
+          std::getline(in, temp);
+          std::getline(in, temp);
+          std::getline(in, temp);
+
+          AssertThrow(min_temp >= 0.0, ExcMessage("Read in of Material header failed (mintemp)."));
+          AssertThrow(delta_temp > 0, ExcMessage("Read in of Material header failed (delta_temp)."));
+          AssertThrow(n_temperature > 0, ExcMessage("Read in of Material header failed (numtemp)."));
+          AssertThrow(min_press >= 0, ExcMessage("Read in of Material header failed (min_press)."));
+          AssertThrow(delta_press > 0, ExcMessage("Read in of Material header failed (delta_press)."));
+          AssertThrow(n_pressure > 0, ExcMessage("Read in of Material header failed (numpress)."));
+
+
+          max_temp = min_temp + (n_temperature-1) * delta_temp;
+          max_press = min_press + (n_pressure-1) * delta_press;
+
+          density_values.reinit(n_temperature,n_pressure);
+          thermal_expansivity_values.reinit(n_temperature,n_pressure);
+          specific_heat_values.reinit(n_temperature,n_pressure);
+          vp_values.reinit(n_temperature,n_pressure);
+          vs_values.reinit(n_temperature,n_pressure);
+          enthalpy_values.reinit(n_temperature,n_pressure);
+          temperature_values.reinit(n_temperature,n_pressure);
+
+          unsigned int i = 0;
+          while (!in.eof())
+            {
+              double unused_value;
+              double rho,alpha,cp,vp,vs,h,T;
+
+              in >> T;
+              if (in.fail())
+                {
+                  in.clear();
+                  T = temperature_values[(i-1)%n_temperature][(i-1)/n_temperature];
+                }
+
+              // Pressure
+              in >> unused_value;
+
+              in >> rho;
+              if (in.fail())
+                {
+                  in.clear();
+                  rho = density_values[(i-1)%n_temperature][(i-1)/n_temperature];
+                }
```

Comment:
Does it make sense to write a small helper function for that, since it's repeated so often? That would also give you the opportunity to explain what exactly it is you're doing here with the `[(i-1)%n_temperature][(i-1)/n_temperature]` part.